R/autoxgboost.R

```diff
@@ -104,11 +106,19 @@ autoxgboost = function(task, measure = NULL, control = NULL, par.set = NULL, max
   }
 
   if (has.cat.feats) {
-    if (factor.encoder == "impact") {
-      base.learner = makeImpactFeaturesWrapper(base.learner)
-    } else {
-      task = createDummyFeatures(task)
+    d = getTaskData(task, target.extra = TRUE)$data
+    feat.cols = colnames(d)[vlapply(d, is.factor)]
+    impact.cols = colnames(d[, vlapply(d, function(x) is.factor(x) && nlevels(x) > impact.encoding.boundary)])
```

Comment:
impact.cols = colnames(d)[vlapply(d, function(x) is.factor(x) && nlevels(x) > impact.encoding.boundary)]